src/main/java/org/jenkinsci/plugins/gitclient/GitToolConfigurator.java

```diff
@@ -0,0 +1,127 @@
+public class GitToolConfigurator extends BaseConfigurator<GitTool> {
+
+    @NonNull
+    @Override
+    public String getName() {
+        return "git";
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Git";
+    }
+
+    @Override
+    public Class getTarget() {
+        return GitTool.class;
+    }
+
+    @Override
+    public boolean canConfigure(Class clazz) {
+        return clazz == GitTool.class;
+    }
+
+    @NonNull
+    @Override
+    public Class getImplementedAPI() {
+        return getTarget();
+    }
+
+    @NonNull
+    @Override
+    public List<Configurator<GitTool>> getConfigurators(ConfigurationContext context) {
+        return Collections.singletonList(this);
+    }
+
+    @NonNull
+    @Override
+    public List<Attribute<GitTool, ?>> getAttributes() {
+        Attribute<GitTool, String> name = new Attribute<>("name", String.class);
+        Attribute<GitTool, String> home = new Attribute<>("home", String.class);
+        Attribute<GitTool, ToolProperty> p = new Attribute<>("properties", ToolProperty.class);
+        p.multiple(true);
+        return Arrays.asList(name, home, p);
+    }
+
+    @Override
+    protected GitTool instance(Mapping mapping, ConfigurationContext context) throws ConfiguratorException {
+        final CNode mproperties = mapping.remove("properties");
+        final String name = mapping.getScalarValue("name");
+        if (JGitTool.MAGIC_EXENAME.equals(name)) {
+            mapping.remove("home"); //Ignored but could be added, so removing to not fail handleUnknown
+            return new JGitTool(instantiateProperties(mproperties, context));
+        } else if (JGitApacheTool.MAGIC_EXENAME.equals(name)) {
+            mapping.remove("home");  //Ignored but could be added, so removing to not fail handleUnknown
+            return new JGitApacheTool(instantiateProperties(mproperties, context));
+        } else {
+            if (mapping.get("home") == null) {
+                throw new ConfiguratorException(this, "Home required for cli git configuration.");
+            }
+            String home = mapping.getScalarValue("home");
+            return new GitTool(name, home, instantiateProperties(mproperties, context));
+        }
+    }
+
+    @Nonnull
+    private List<ToolProperty<?>> instantiateProperties(@CheckForNull CNode props, @Nonnull ConfigurationContext context) throws ConfiguratorException {
+        final Configurator<ToolProperty> configurator = context.lookupOrFail(ToolProperty.class);
+        List<ToolProperty<?>> toolProperties = new ArrayList<>();
+        if (props == null) {
+            return toolProperties;
+        }
+        if (props instanceof Sequence) {
+            Sequence s = (Sequence) props;
+            for (CNode cNode : s) {
+                toolProperties.add(configurator.configure(cNode, context));
+            }
+        } else {
+            toolProperties.add(configurator.configure(props, context));
+        }
```

Comment:
This if is a bit unclear for me. IIUC, the `if` clause is when there are more than one element in the list, so you're looping over them and using each element as input. So far, it's clear for me. But in the `else` clause, you're using the `props` element as input. Shouldn't it be the content of that object? Something equivalent to `list.get(0)`

Reply:
I think it shouldn't since the configuration is a yaml file, but I son't think it hurts, so fine for me